OvmfPkg/IoMmuDxe/AmdSevIoMmu.c

```diff
@@ -513,6 +884,83 @@ IoMmuUnmap (
            );
 }
 
+/**
+ * Allocate CommonBuffer from pre-allocated shared memory.
+ *
+ * @param MemoryType          Memory type
+ * @param CommonBufferPages   Pages of CommonBuffer
+ * @param PhysicalAddress     Allocated physical address
+ * @param ReservedMemBitmap   Bitmap which indicates the allocation of reserved memory
+ * @return EFI_SUCCESS  Successfully allocate CommonBuffer
+ */
+STATIC
+EFI_STATUS
+AllocateCommonBuffer (
+  IN EFI_MEMORY_TYPE        MemoryType,
+  IN UINTN                  CommonBufferPages,
+  OUT EFI_PHYSICAL_ADDRESS  *PhysicalAddress,
+  OUT UINT32                *ReservedMemBitmap
+  )
+{
+  EFI_STATUS  Status;
+
+  Status = InternalAllocateBuffer (
+             AllocateMaxAddress,
+             MemoryType,
+             CommonBufferPages,
+             ReservedMemBitmap,
+             PhysicalAddress
+             );
+  ASSERT (Status == EFI_SUCCESS);
+
+  mReservedMemBitmap |= *ReservedMemBitmap;

```

Comment:
I am thinking it better let caller to do this instead of here, since mReservedMemBitmap is a module static variable, this assignment make this function tight coupled with other modular variable.

Reply:
The reserved memory related functions and data structure will be moved a separate file. mReservedMemBitmap will be in that file as well. See https://github.com/tianocore/edk2/pull/3709